Server/src/quadradiusr_server/auth.py

```diff
@@ -1,52 +1,53 @@
+import base64
+import hashlib
+import os
 import uuid
 from datetime import datetime, timedelta, timezone
 from typing import Optional
 
 import jwt
 
 from quadradiusr_server.config import AuthConfig
-
-
-class User:
-    def __init__(self, id_: str) -> None:
-        self._id = id_
-
-    @property
-    def id_(self):
-        return self._id
-
-    def to_json(self):
-        return {
-            'id': self.id_,
-        }
-
-    @classmethod
-    def from_json(cls, json):
-        return User(json['id'])
+from quadradiusr_server.db.base import User
+from quadradiusr_server.db.repository import Repository
 
 
 class Auth:
-    def __init__(self, config: AuthConfig) -> None:
+    def __init__(self, config: AuthConfig, repository: Repository) -> None:
         self.config = config
+        self.repository = repository
         self.__secret = str(uuid.uuid4())
         self.__algorithm = 'HS256'
 
+    async def login(self, username: str, password: bytes) -> Optional[User]:
+        user_repository = self.repository.user_repository
+        user: User = await user_repository.get_by_username(username)
+        if not user:
+            return None
+
+        if check_password(password, user.password_):
```

Comment:
For function returning bool I'd personally use a name signifying that, like is_pasword_valid